core/src/main/scala/org/apache/spark/HttpFileServer.scala

```diff
@@ -50,6 +50,15 @@ private[spark] class HttpFileServer(
 
   def stop() {
     httpServer.stop()
+    
+    // If we only stop sc, but the driver process still run as a services then we need to delete 
+    // the tmp dir, if not, it will create too many tmp dirs
+    try {
+      Utils.deleteRecursively(baseDir)
```

Comment:
This seems OK to me. I would make the error log a warning though as it's non-fatal. Consider using an interpolated string here as you do below.